src/utils/validadeMediaAndText/index.spec.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  validateImageAspectRatio,
+  validateImageResolution,
+  validateMediaSize,
+  validateTextLength,
+  validateVideoDuration,
+  validateVideoResolution,
+} from '.';
+
+const mockFileImage = new File([new ArrayBuffer(10 * 10 + 1)], 'mockfile.jpg', {
+  type: 'image/jpeg',
+});
+
+const mockFileVideo = new File([new ArrayBuffer(10 * 10 + 1)], 'mockfile.mp4', {
+  type: 'video/mp4',
+});
+
+describe('validate texts', () => {
+  it('validate text length', () => {
+    const limiteLength = 2000;
+
+    expect(validateTextLength('Bom dia!', limiteLength)).toBe(true);
+  });
+});
+
+describe('validate medias', () => {
+  it('validate media size', () => {
+    const limitSize = 9999;
+
+    expect(validateMediaSize(mockFileImage, limitSize)).toBe(true);
+  });
+});
+
+describe('validate images', () => {
+  beforeEach(() => {
+    let img = '';
+    global.Image = vi.fn().mockImplementation(() => ({
+      addEventListener: (event: string, cb: () => void): void => {
+        const intervalId = setInterval(() => {
+          if (event === 'load' && img) {
+            cb();
+            clearInterval(intervalId);
+          }
+        }, 100);
+      },
+      set src(value: string) {
+        setTimeout(() => {
+          img = value;
+        }, 1000);
+      },
+    }));
+  });
+
+  it('validate image resolution', async () => {
+    const limitWidth = 1920;
+    const limitHeight = 1080;
+
+    expect(
+      await validateImageResolution(mockFileImage, limitWidth, limitHeight)
+    ).toBe(false);
+    await expect(
+      validateImageResolution(mockFileVideo, limitWidth, limitHeight)
+    ).rejects.toThrow('File is not a image');
```

Comment:
To throw? Acho que não, isso não é um erro.

Um erro é quando algo acontece dentro do nosso app que não esperamos e não temos caso de tratamento, isso é um erro. Nesse caso é apenas um check fail pq o arquivo é muito grande, vc precisa retornar algo como `false` ou coisa parecida

Talvez o nome do método poderia ser `isImageResolutionValid()` ou `checkImageResolution()` ou algo assim, retornando apenas um `true` ou `false` de se é valido ou nao.